packages/next/build/index.ts

```diff
@@ -1267,21 +1264,17 @@ export default async function build(
                       )
                     : appPaths?.find((p) => p.startsWith(actualPage + '/page.'))
 
-                const pageRuntime =
+                const staticInfo =
                   pagesDir && pageType === 'pages' && pagePath
-                    ? (
-                        await getPageStaticInfo({
-                          pageFilePath: join(pagesDir, pagePath),
-                          nextConfig: config,
-                        })
-                      ).runtime
-                    : undefined
-
-                if (hasServerComponents && pagePath) {
-                  if (isServerComponentPage(config, pagePath)) {
-                    isServerComponent = true
-                  }
-                }
+                    ? await getPageStaticInfo({
+                        pageFilePath: join(pagesDir, pagePath),
+                        nextConfig: config,
+                      })
+                    : {}
+                const pageRuntime = staticInfo.runtime
+                isServerComponent =
+                  pageType === 'app' &&
+                  staticInfo.rsc !== RSC_MODULE_TYPES.client
```

Comment:
I think it's more readable to use the equality operator directly:

```suggestion
                  staticInfo.rsc === RSC_MODULE_TYPES.server
```